corehq/apps/data_dictionary/models.py

```diff
@@ -55,6 +55,9 @@ def get_or_create(cls, name, case_type, domain):
                     name=name, case_type__name=case_type, case_type__domain=domain
                 )
             except CaseProperty.DoesNotExist:
-                case_type_obj = CaseType.objects.get(domain=domain, name=case_type)
+                try:
+                    case_type_obj = CaseType.objects.get(domain=domain, name=case_type)
+                except CaseType.DoesNotExist:
+                    case_type_obj = CaseType.objects.create(domain=domain, name=case_type)
```

Comment:
is this the only place where we get or create a CaseType? if not we should also wrap this operation in its own CriticalSection.

actually for cleanliness it might be worth adding a get_or_create method to CaseType just like you did here, which uses its own CriticalSection